ios/FluentUI/Button/Button.swift

```diff
@@ -396,33 +391,41 @@ open class Button: UIButton {
         isAdjustingCustomContentEdgeInsetsForImage = false
     }
 
-    private func updateBackgroundColor() {
-        guard let window = window else {
-            return
+    private func primaryFilledBackgroundColor() -> UIColor {
+        if isHighlighted {
+            return UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.brandBackground1Pressed])
         }
 
+        if isFocused {
+            return UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.brandBackground1Selected])
+        }
+
+        return UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.brandBackground1])
+    }
+
+    private func primaryDangerFilledBackgroundColor() -> UIColor {
```

Comment:
Does this need to be a function, or could it be a `var` with a custom getter?

```swift
private var primaryDangerFilledBackgroundColor: UIColor {
    // TODO: in the future, we want to have highlighted state defined for danger buttons.
    // For now, highlighted/isPressed are not differentiated for danger buttons.
    return UIColor(dynamicColor: fluentTheme.aliasTokens.sharedColors[.dangerBackground2])
}

Reply:
I think the main reason it'd be a function is if it takes any arguments, otherwise we prefer to use vars.